### PR TITLE
[AGW][C/C++] Clean up CmakeLists and fix MME optimization for release builds

### PR DESCRIPTION
**Pull request by themarwhal**
Signed-off-by: Marie Bremner <marwhal@fb.com>

## Summary
* Using `CMAKE_LINKER_FLAGS_*` -> `CMAKE_EXE_LINKER_FLAGS_*` (for session_manager/sctpd/connection_tracker)
	* Noticed that Visual Studio was not highlighting `CMAKE_LINKER_FLAGS_*` and from a quick google search, they seem to both be valid. (Although, `CMAKE_EXE_LINKER_FLAGS_*` seems to be more common.) But since MME uses `CMAKE_EXE_LINKER_FLAGS_*` modifying the rest to be consistent across the project. 
* MME is overwriting it's -O2 optimization level for release builds due to a typo. Fixing it as I don't think it was intentional. (But maybe it was? Requesting input from @ulaskozat + @ardzoht )
  * https://gist.github.com/themarwhal/68957d37b7e0fbfec5353cb9dd3187f2 (This is the verbose build output for release builds on master)

## Notes
Just noting here that although the default optimization level for Debug builds is -O1, we manually set it to -O0. 

## Test Plan
build for debug + release

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Review by uri200 — APPROVED** [review]

File: lte/gateway/c/connection_tracker/CMakeLists.txt [hunk 1]
```diff
@@ -17,26 +17,27 @@ add_compile_options(-std=c++14)
 
 include($ENV{MAGMA_ROOT}/orc8r/gateway/c/common/CMakeProtoMacros.txt)
 
+set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -g -O1")
```
Comment (themarwhal) [inline lte/gateway/c/connection_tracker/CMakeLists.txt]: note for monday : remove

File: lte/gateway/c/sctpd/CMakeLists.txt [hunk 2]
```diff
@@ -3,11 +3,13 @@ set(CMAKE_CXX_STANDARD 14)
 
 project(MagmaSctpd)
 
+set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -g -O1")
```
Comment (themarwhal) [inline lte/gateway/c/sctpd/CMakeLists.txt]: note for monday : remove

File: lte/gateway/c/session_manager/CMakeLists.txt [hunk 3]
```diff
@@ -20,17 +20,19 @@ include(FindClangFormat)
 
 include($ENV{MAGMA_ROOT}/orc8r/gateway/c/common/CMakeProtoMacros.txt)
 
+set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -g -O1")
```
Comment (themarwhal) [inline lte/gateway/c/session_manager/CMakeLists.txt]: note for monday : remove

**Comment by themarwhal** [comment]
Closing for now as I'd like to do more experiments to get measurements

**Pull request closed without merging**